modules/renter/upload.go

```diff
@@ -45,15 +45,31 @@ var (
 	}()
 )
 
+// validateSiapath checks that a Siapath is a legal filename.
+// ../ is disallowed to prevent directory traversal,
+// and paths must not begin with / or be empty.
+func validateSiapath(siapath string) error {
+	if strings.HasPrefix(siapath, "/") || strings.HasPrefix(siapath, "./") {
+		return errors.New("nicknames cannot begin with /")
+	}
+
+	if siapath == "" {
+		return ErrEmptyFilename
+	}
+
+	if strings.Contains(siapath, "../") {
+		return errors.New("directory traversal is not allowed")
+	}
```

Comment:
A cleaner way might be:
```go
if path.Clean(siapath) != siapath {
    return errors.New("illegal path")
}
```
however, this will also reject paths like `///foo///bar` (-> `/foo/bar`). Not sure if this is desirable.